docs/embed-c.md

```diff
@@ -1,4 +1,112 @@
 # Embedding Wasmtime in C
 
-... more coming soon
+This document shows an example of how to embed Wasmtime using the [C API](https://github.com/WebAssembly/wasm-c-api) to execute a simple wasm program.
 
+## Hello, world!
+
+An excellent example of C API usage can be found at https://github.com/WebAssembly/wasm-c-api/blob/master/example/hello.c
+
+### WebAssembly code
+
+```wat
+(module
+  (func $hello (import "" "hello"))
+  (func (export "run") (call $hello))
+)
+```
+
+### C code
+
+The wasmtime engine and its storage has to be initialized, before the WebAssembly modules can be loaded:
+
+```C
+  wasm_engine_t* engine = wasm_engine_new();
+  wasm_store_t* store = wasm_store_new(engine);
+```
+
+The WebAssembly module bytecode can be loaded into memory and has to be provided as `wasm_byte_vec_t` to be compiled:
+
+```C
+  own wasm_module_t* module = wasm_module_new(store, &binary);
+  if (!module) {
+    printf("> Error compiling module!\n");
+    return 1;
+  }
+```
```

Comment:
We should mention the `own` macro for people not familiar with the convention. Maybe "`own` is a macro that doesn't actually do anything, but helps document which pointers "own" the resources associated with the objects they point to" or so.

Reply:
FWIW I'd recommend just not using it entirely in the introductory documentation. It feels like an unnecessary speed bump that otherwise can be explained later.